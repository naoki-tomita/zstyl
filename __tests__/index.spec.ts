import { parseStyle, random, toStyleString } from ".."

describe("zstyl", () => {
  describe("toStyleString", () => {
    it("should create zheleznaya styled component", () => {
      const styleString = toStyleString("id", { color: "green" })`
        display: flex;
        justify-content: center;
        color: ${({ color }) => color};

        &:hover {
          color: red;
        }

        &:active {
          color: blue;
        }

        .hoge {
          width: 100%;
        }
      `;
      expect(styleString).toMatchSnapshot();
    });
  });

  describe("random", () => {
    it("should include '9' and 'a'", () => {
      const actual = random(1000);
      expect(actual).toContain("9");
      expect(actual).toContain("a");
    });
  });
})
