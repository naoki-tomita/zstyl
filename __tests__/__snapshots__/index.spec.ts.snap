// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zstyl toStyleString should create zheleznaya styled component 1`] = `
"*[data-zstyl='id'] { display: flex; justify-content: center; color: green; }
*[data-zstyl='id']:hover { color: red; }
*[data-zstyl='id']:active { color: blue; }"
`;
